### PR TITLE
Document BYTES row format

### LINKED ISSUE #1028: Document `BYTES` as a new data format
### Related code PR

https://github.com/risingwavelabs/risingwave/pull/10592

### Which part(s) of the docs might be affected or should be updated? And how?

Document `BYTES` as a new data format. Currently only Kafka source is tested, confirm if other sources also support BYTES data format. 

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Document BYTES row format under CREATE SOURCE and Ingest data from Kafka.

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/10592

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1028

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/commands/sql-create-source.md [hunk 1]
```diff
@@ -271,6 +271,12 @@ FORMAT UPSERT
 ENCODE AVRO
 ```
 
+### Bytes
+
+Currently, we require users to decode the stream data through a certain format. But the user may want no decoding phase. This is a missing capability, which is fundamental, although uncommon to appear in real-world use cases. We only accept table with exactly one field of BYTEA type in ROW FORMAT BYTES. Only append-only type is considered here because BYTEA cannot be the primary key for the table #10577.
+
+If
```
Comment (wugouzi) [inline docs/sql/commands/sql-create-source.md]: what is this "if"?

Comment (emile-00) [reply docs/sql/commands/sql-create-source.md]: Sorry about that! Seems like I made an error when creating the pull request. This isn't the intended description. Please take a look at the updated version. Thank you!

File: docs/sql/commands/sql-create-source.md [hunk 2]
```diff
@@ -271,6 +271,15 @@ FORMAT UPSERT
 ENCODE AVRO
 ```
 
+### Bytes
+
+RisingWave allows you to read data streams without decoding the data by using the `BYTES` row format. However, the table or source can have exactly one field of `BYTEA` data and it cannot be the primary key.
```
Comment (wugouzi) [inline docs/sql/commands/sql-create-source.md]: It can be the primary key.

**Comment by wugouzi** [comment]
Rest LGTM. Thanks

**Review by wugouzi — APPROVED** [review]

**Review by hengm3467 — APPROVED** [review]

**Pull request merged**